src/data/list/indexes.lean

```diff
@@ -0,0 +1,168 @@
+import data.list.basic
+import data.list.defs
+import data.list.range
+import data.list.zip
+import logic.basic
+
+universes u v
+
+open function
+
+namespace list
+
+variables {α : Type u} {β : Type v}
+
+
+section foldr_with_index
+
+/-- Specification of `foldr_with_index_aux`. -/
+def foldr_with_index_aux_spec (f : ℕ → α → β → β) (start : ℕ) (b : β)
+  (as : list α) : β :=
+foldr (uncurry f) b $ zip (range' start as.length) as
+
+theorem foldr_with_index_aux_spec_cons (f : ℕ → α → β → β) (start b a as)
+  : foldr_with_index_aux_spec f start b (a :: as)
+  = f start a (foldr_with_index_aux_spec f (start + 1) b as) := rfl
+
+theorem foldr_with_index_aux_correct (f : ℕ → α → β → β) (start b as)
+  : foldr_with_index_aux f start b as = foldr_with_index_aux_spec f start b as :=
```

Comment:
We don't usually use this `_correct` naming scheme.

Reply:
Just say more in the name, e.g. along the lines of `X_eq_Y`.

Reply:
Fixed throughout.